corehq/apps/translations/views.py

```diff
@@ -101,11 +101,11 @@ def _generate_translations_for_po(self, worksheet):
     def _generate_po_file(self, worksheet):
         """
         extract translations from worksheet and converts to a po file
-        :return: PoFileGenerator object
+        :return: list of files generated
         """
         translations = self._generate_translations_for_po(worksheet)
-        po_file_generator = PoFileGenerator(translations, {})
-        return po_file_generator
+        with PoFileGenerator(translations, {}) as po_file_generator:
+            return po_file_generator.generated_files[0][1]
```

Comment:
Unrelated question: use of index constants in `.generated_files[0][1]` is a bit magic-y. Why only process the first file? If there is only ever one, could `.generated_files` be refactored to be `.generated_file`?

Finally, named tuple would be nice for readability here, making it `.generated_files[0].path` with the named tuple defined as
```py
from collections import namedtuple

PoFileInfo = namedtuple("PoFileInfo", "name path")
```
and

https://github.com/dimagi/commcare-hq/blob/7d42f35fb4d4305ec1c75131f6808639cb74b1ef/corehq/apps/app_manager/app_translations/generators.py#L228

would change to
```py
self.generated_files.append(PoFileInfo(file_name, temp_file.name))
```

Reply:
So,
`PoFileGenerator` has the ability to generate multiple files. This django view however just picks the first sheet of the excel file uploaded because of the workflow after this process explained below. One can definitely modify this to process all sheets in the uploaded excel, generate a bunch of PO files, zip them and return that to the user. But currently there is not a use case with multiple sheets.

A little more context,
The way its used is that one needs to convert an excel sheet(not workbook) to a PO file and then upload it to Transifex manually via their interface. Till now we have always done one file at a time. I am open to making it do what i said above, just didn't see the use for stretching it so far yet.

Reply:
Am I correct in understanding that `po_file_generator.generated_files` will contain files for every sheet even though only the first one will be used? Seems like it's doing more work than is necessary. I suppose that's a reasonable implementation detail if performance of processing all sheets is not a problem.

The namedtuple would still be nice.

Reply:
> Am I understanding correctly that po_file_generator.generated_files will contain a file for every sheet even though only the first one will be used?

The view picks only the first sheet and gives that to `PoFileGenerator`, as implemented [here](https://github.com/dimagi/commcare-hq/blob/15f567011ad02213351f9bc7d14ca907c35e1fc5/corehq/apps/translations/views.py#L131-L130).

> The namedtuple would still be nice.

Yup! made that change. Could not push because of internet :) .You should be able see the change shortly.